src/frontend/screens/Success.tsx

```diff
@@ -48,16 +46,17 @@ export const Success = ({
       </View>
       <Button
         fullWidth
+        // due to conditional rendering, there is a brief second, where the state is no longer pending, but the user has not been navigated. So if we don't do catch all these state, the button flickers briefly before navigating
```

Comment:
just to note, this flickering only happens during conditinal rendering. If we were to do:

```ts
const {mutation} = useSomeReactQueryMutation()

mutate(val, {
    onSuccess:() => navigate("nextPage")
})

```
this onsucess callback is called immediately and we wont see any flickering